x/incentive/keeper/update.go

```diff
@@ -156,27 +156,32 @@ func (k Keeper) updatePrograms(ctx sdk.Context) error {
 
 // EndBlock updates incentive programs and reward accumulators, then sets LastRewardTime
 // to the current block time. Also protects against negative time elapsed (without causing chain halt).
-func (k Keeper) EndBlock(ctx sdk.Context) error {
+// In addition to regular error, returns a boolean indicating whether the main logic was skipped due
+// to a blockTime issue. These situations are accompanied by error logs.
+func (k Keeper) EndBlock(ctx sdk.Context) (err error, skipped bool) {
 	blockTime := ctx.BlockTime().Unix()
 	if blockTime < 0 {
 		k.Logger(ctx).Error(
 			incentive.ErrDecreaseLastRewardTime.Error(),
 			"negative block time", blockTime,
 		)
-		return nil
+		return nil, true
```

Comment:
I think this should be an error

Reply:
This logs an error, but does not return one. Returned errors become panic (chain halt)

Reply:
well, I would say that this is an impossible scenario, to have ctx.BlockTime before 1971